src/Commands/Auth/LoginCommand.php

```diff
@@ -33,62 +35,73 @@ class LoginCommand extends TerminusCommand
      * @usage --machine-token=<machine_token> Logs in a user granted the machine token <machine_token>.
      * @usage Logs in a user with a previously saved machine token.
      * @usage --email=<email> Logs in a user with a previously saved machine token belonging to <email>.
+     *
+     * @throws \Pantheon\Terminus\Exceptions\TerminusNotFoundException
+     * @throws TerminusException
      */
-    public function logIn(array $options = ['machine-token' => null, 'email' => null,])
+    public function logIn(array $options = ['machine-token' => null, 'email' => null,]): void
     {
         $tokens = $this->session()->getTokens();
 
-        if (isset($options['machine-token']) && !is_null($token_string = $options['machine-token'])) {
+        if (isset($options['machine-token'])) {
             try {
-                $token = $tokens->get($token_string);
+                $token = $tokens->get($options['machine-token']);
+                $this->processLogIn($token);
+                return;
             } catch (\Exception $e) {
                 $this->log()->notice('Logging in via machine token.');
-                $tokens->create($token_string);
+                $tokens->create($options['machine-token']);
```

Comment:
Was this use case broken before? Is it also broken on the 2.x branch? New code looks correct.

Reply:
@greg-1-anderson  No, just refactored to get rid of unnecessary vars and checks

Reply:
@greg-1-anderson not sure about `2.x`, need to check the code